aviary/examples/external_subsystems/engine_NPSS/NPSS_Model/DesignEngineGroup.py

```diff
@@ -0,0 +1,167 @@
+import numpy as np
+import openmdao.api as om
+import subprocess as subprocess
+from openmdao.utils.file_wrap import FileParser
+from aviary.subsystems.propulsion.engine_sizing import SizeEngine
+from aviary.utils.functions import get_path
+import time
+import os
+
+from aviary.examples.external_subsystems.engine_NPSS.engine_variables import Aircraft, Dynamic
+
+# Number of points in NPSS model deck.
+# Will need to be updated if the size of the deck changes.
+vec_size = 72
+
+
+class NPSSExternalCodeComp(om.ExternalCodeComp):
+    def setup(self):
+        self.add_input('Alt_DES', val=0.0, units='ft', desc='design altitude')
+        self.add_input('MN_DES', val=0.0, units=None, desc='design Mach number')
+        self.add_input('W_DES', val=240.0, units='lbm/s', desc='design mass flow')
+
+        self.add_output('Fn_SLS', val=1.0, units='lbf',
+                        desc='net thrust at sea-level-static conditions')
+        self.add_output('Wf_training_data', val=np.ones(vec_size),
+                        units='lbm/s', desc='fuel flow training data')
+        self.add_output('thrust_training_data', val=np.ones(
+            vec_size), units='lbf', desc='thrust training data')
+        self.add_output('thrustmax_training_data', val=np.ones(vec_size),
+                        units='lbf', desc='maximum thrust training data')
+
+        self.input_file = get_path(
+            './examples/external_subsystems/engine_NPSS/NPSS_Model/Design_files/input.int')
+        self.output_file = get_path(
+            './examples/external_subsystems/engine_NPSS/NPSS_Model/Design_files/output.int')
+
+        self.options['external_input_files'] = [self.input_file]
+        self.options['external_output_files'] = [self.output_file]
+
+        # self.options['command'] = ['c:/NPSS.nt.ver32_VC14_64/bin/npss.nt.exe','turbojet.run']
+
+        run_location = get_path(
+            './examples/external_subsystems/engine_NPSS/NPSS_Model/turbojet.run')
+        self.options['command'] = ['runnpss', run_location]
+
+    def setup_partials(self):
+        # this external code does not provide derivatives, use finite difference
+        # Note: step size should be larger than NPSS solver tolerance
+        self.declare_partials(of='*', wrt='*', method='fd', step=1e-3)
+
+    def compute(self, inputs, outputs):
+        Alt_DES = inputs['Alt_DES']
+        MN_DES = inputs['MN_DES']
+        W_DES = inputs['W_DES']
+
+        # generate the input file for the external code
+        with open(self.input_file, 'w') as input_file:
```

Comment:
Is an in-memory wrap of NPSS feasible in the future?

Reply:
I think looking into that would be a great next step, if getting the builder to run faster becomes a priority.